src/main/java/org/disq_bio/disq/HtsjdkReadsRdd.java

```diff
@@ -0,0 +1,33 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import org.apache.spark.api.java.JavaRDD;
+
+/**
+ * A {@link HtsjdkReadsRdd} is the distributed equivalent of a htsjdk {@link
+ * htsjdk.samtools.SamReader}. It represents a SAM, BAM, or CRAM file stored in a distributed
+ * filesystem, and encapsulates a Spark RDD containing the reads in it.
+ *
+ * <p>Use a {@link HtsjdkReadsRddStorage} to read and write {@link HtsjdkReadsRdd}s.
+ *
+ * @see HtsjdkReadsRddStorage
+ */
+public class HtsjdkReadsRdd {
```

Comment:
Why does it requires the prefix `Htsjdk` (here and other code in this package)? If we said that this library targets to use the htsjdk library interfaces (in the future, I hope that the proper interfaces in htsjdk-next), we do not require this prefix.

Reply:
I added it because of a discussion we had a while back about using different data models (i.e. non-htsjdk) for reads and variants. Interested to see what others think too.

Reply:
If changed to `RecordsRdd<Metadata, Record>`, the implementation can either maintain the prefix or be called `SAMRecordRdd`.

Reply:
I'd prefer to see the `Htsjdk` prefixes not be in the top level of the API as well. The metadata and record types can be generic as discussed here and in some other comments. Not that it should block the initial commit, but I do like the idea. Also, is it worth abstracting RDD to a type parameter at top level to ease migration to DataSet ? It would help with unifying interface.

Reply:
Please keep the `Htsjdk` prefixes for now.